Updated documentation to be easier for people to go through it and ge…

mhedgpeth opened the pull request:
…t set up quickly

This is from the discussion in issue #20. I'm very new to azure so I took some liberties to explain some things that were confusing to me as I went through the process.

g0atcheez commented:
There is a slight typo in your documentation. #21 

> +2. Create a new AAD application ```#azureAdApplication = New-AzureRmApplication -DisplayName "Test Kitchen" -HomePage "http://www.yourcompany.com" -IdentifierUris "http://www.yourcompany.com" -Password PutSomethingHere```

The command is New-AzureRmAdApplication

mhedgpeth commented:
I fixed the typo, corrected another issue with the rendering, and added some context for people who are truly new to azure. I think that this will be used by people who want to use test kitchen, have a windows background, but have never used azure before.

stuartpreston commented on README.md:
````diff
@@ -16,11 +16,27 @@ Note if you are running the ChefDK you may need to prefix the command with chef,
-Using a text editor, open or create the file ```~/.azure/credentials``` and add the following section, noting there is one section per Subscription ID.  **Make sure you save the file with UTF-8 encoding** 
+For the driver to interact with the Microsoft Azure Resource management REST API, a Service Principal needs to be configured with Contributor rights against the specific subscription being targeted.  Using an Organizational (AAD) account and related password is no longer supported.  To create a Service Principal and apply the correct permissions, follow the below instructions (taken from the 'Aut ...
+
+1. Ensure that [the Azure CLI is installed](https://azure.microsoft.com/en-us/documentation/articles/xplat-cli-install/)
````
The rest of the instructions that were inserted are for PowerShell, so Azure-cli is not required in that instance...

stuartpreston commented on README.md:
````diff
@@ -16,11 +16,27 @@ Note if you are running the ChefDK you may need to prefix the command with chef,
+1. Ensure that [the Azure CLI is installed](https://azure.microsoft.com/en-us/documentation/articles/xplat-cli-install/)
+2. Ensure that you have an active Azure subscription (you can get started [for free](https://azure.microsoft.com/en-us/free/) or use your [MSDN Subscription](https://azure.microsoft.com/en-us/pricing/member-offers/msdn-benefits/))
+3. In PowerShell, log into Azure with ```Login-AzureRmAccount```
+4. Create a new AAD application ```#azureAdApplication = New-AzureRmAdApplication -DisplayName "Test Kitchen" -HomePage "http://www.yourcompany.com" -IdentifierUris "http://www.yourcompany.com" -Password PutSomethingHere```
````
I don't think the # is required...?

stuartpreston commented on README.md:
```diff
@@ -16,11 +16,27 @@ Note if you are running the ChefDK you may need to prefix the command with chef,
+You are now ready to configure kitchen-azurerm to use these credentials. You will use four elements taken from the steps above:
+1. **Subscription ID**: listed after the command in Step #7 above
+2. **Client ID**: this will be the ApplicationId from the application created in Step #2 above
+3. **Client Secret/Password**: this will be the password you supplied in the command in Step #2 above
```
Don't you mean from Step 4?

stuartpreston commented:
@mhedgpeth Thanks for the PR. I've added some comments that may need looking at, also if we're going to add the explicit commands to set up the service principal rather than link to the official docs, then I feel that we should include both Azure-cli and PowerShell versions in line with the spirit of the project, as many users of the driver will not be on a Windows workstation.  (Though I'm happy to add the Azure-cli bits if you want)

mhedgpeth commented:
I didn't realize there were azure-cli versions of those commands. Shouldn't
the documentation just document those? That way it would work for everyone.

On Fri, Feb 26, 2016 at 3:48 PM Stuart Preston <notifications@github.com>
wrote:

> @mhedgpeth <https://github.com/mhedgpeth> Thanks for the PR. I've added
> some comments that may need looking at, also if we're going to add the
> explicit commands to set up the service principal rather than link to the
> official docs, then I feel that we should include both Azure-cli and
> PowerShell versions in line with the spirit of the project, as many users
> of the driver will not be on a Windows workstation. (Though I'm happy to
> add the Azure-cli bits if you want)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pendrica/kitchen-azurerm/pull/21#issuecomment-189494031>
> .
>

stuartpreston commented:
Exactly. That's why the link to the documentation pointed at the azure-cli version of the instructions...




On Fri, Feb 26, 2016 at 4:12 PM -0800, "mhedgpeth" <notifications@github.com<mailto:notifications@github.com>> wrote:

I didn't realize there were azure-cli versions of those commands. Shouldn't
the documentation just document those? That way it would work for everyone.

On Fri, Feb 26, 2016 at 3:48 PM Stuart Preston <notifications@github.com>
wrote:

> @mhedgpeth <https://github.com/mhedgpeth> Thanks for the PR. I've added
> some comments that may need looking at, also if we're going to add the
> explicit commands to set up the service principal rather than link to the
> official docs, then I feel that we should include both Azure-cli and
> PowerShell versions in line with the spirit of the project, as many users
> of the driver will not be on a Windows workstation. (Though I'm happy to
> add the Azure-cli bits if you want)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pendrica/kitchen-azurerm/pull/21#issuecomment-189494031>
> .
>

—
Reply to this email directly or view it on GitHub<https://github.com/pendrica/kitchen-azurerm/pull/21#issuecomment-189533718>.

mhedgpeth commented:
Ah, then that's part of my issue because I got lost (that documentation
gives you a few versions of doing it). Let me check it again and then
update the PR. Thanks for your patience with it.

Michael

On Fri, Feb 26, 2016 at 6:13 PM Stuart Preston <notifications@github.com>
wrote:

> Exactly. That's why the link to the documentation pointed at the azure-cli
> version of the instructions...
>
>
>
>
> On Fri, Feb 26, 2016 at 4:12 PM -0800, "mhedgpeth" <
> notifications@github.com<mailto:notifications@github.com>> wrote:
>
> I didn't realize there were azure-cli versions of those commands. Shouldn't
> the documentation just document those? That way it would work for everyone.
>
> On Fri, Feb 26, 2016 at 3:48 PM Stuart Preston <notifications@github.com>
> wrote:
>
> > @mhedgpeth <https://github.com/mhedgpeth> Thanks for the PR. I've added
> > some comments that may need looking at, also if we're going to add the
> > explicit commands to set up the service principal rather than link to the
> > official docs, then I feel that we should include both Azure-cli and
> > PowerShell versions in line with the spirit of the project, as many users
> > of the driver will not be on a Windows workstation. (Though I'm happy to
> > add the Azure-cli bits if you want)
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/pendrica/kitchen-azurerm/pull/21#issuecomment-189494031
> >
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/pendrica/kitchen-azurerm/pull/21#issuecomment-189533718
> >.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pendrica/kitchen-azurerm/pull/21#issuecomment-189533906>
> .
>

mhedgpeth commented:
I'm simplified it more after understanding things a bit better. The problem I had was that I got lost in the documentation and then didn't know what values I needed. I added some help text to help people go in the right direction with it, and eliminated the steps as they are redundant.

stuartpreston commented:
:+1: ignoring the Rubocop error in the build which is nothing to do with this documentation update.
